server/src/test/java/org/cloudfoundry/identity/uaa/provider/token/JwtBearerAssertionTokenAuthenticatorTest.java

```diff
@@ -136,37 +171,37 @@ public String getPublicKey(final String tenantId, final String deviceId) throws
 
     @Test(expected = AuthenticationException.class)
     public void testNonExistentClient() {
-        String token = new MockAssertionToken().mockAssertionToken(DEVICE_1_ID, System.currentTimeMillis() - 240000,
-                600, TENANT_ID, AUDIENCE);
+        String token = new MockAssertionToken().mockAssertionToken(DEVICE_1_CLIENT_ID, DEVICE_1_ID,
+                System.currentTimeMillis(), 600, TENANT_ID, AUDIENCE);
         
         //remove d1 client from mock
-        when(this.clientDetailsService.loadClientByClientId(DEVICE_1_ID))
+        when(this.clientDetailsService.loadClientByClientId(DEVICE_1_CLIENT_ID))
         .thenReturn(null);
         this.tokenAuthenticator.setClientDetailsService(this.clientDetailsService);
```

Comment:
seems worthwhile in a separate unit test.   this method does declare a runtime exception on the interface.